test/common/config/BUILD

```diff
@@ -273,7 +272,6 @@ envoy_cc_test(
 envoy_cc_test(
     name = "subscription_impl_test",
     srcs = ["subscription_impl_test.cc"],
-    tags = ["fails_on_windows"],
     deps = [
```

Comment:
This appears to be flaky, on Windows, if you could run this through a number of times (not using cached test results.)

It doesn't seem that we want to toggle flaky = True for linux or macOS, and I'm wondering if it's worth enabling at this time until we can figure out how the underlying event signaling of the windows implementation is triggering extra mock events.

Reply:
Both these tests succeed 100% of the time when run many times on my machine, but if they are failing for others I can close this PR. A weird thing that I noticed is that I can't seem to run all of the config tests at once. Here are the bazel commands I'm using:

```
bazel --output_base=c:/_eb test -c opt --config=msvc-cl --show_task_finish --verbose_failures --test_output=all //test/common/config:subscription_impl_test --cache_test_results=no
// test passes

bazel --output_base=c:/_eb test -c opt --config=msvc-cl --show_task_finish --verbose_failures --test_output=all //test/common/config:filesystem_subscription_impl_test --cache_test_results=no
// test passes

bazel --output_base=c:/_eb test -c opt --config=msvc-cl --show_task_finish --verbose_failures --test_output=all //test/common/config... --cache_test_results=no
ERROR: Skipping '//test/common/config...': no such package 'test/common/config...': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - C:/users/administrator/envoy/test/common/config...
ERROR: no such package 'test/common/config...': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - C:/users/administrator/envoy/test/common/config...
INFO: Elapsed time: 1.540s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Reply:
@nigriMSFT youre missing a slash in `//test/common/config...`, try with `//test/common/config/...`

Reply:
doi

Reply:
@nigriMSFT Maybe leave this open until https://github.com/envoyproxy/envoy/pull/11107 is merged (and then merge master), so we can reproduce in CI as well as locally. If we can demonstrate flakiness we can also try to make the tests more robust.

Reply:
When running all config tests at once, I see the same flakiness described. I'll close this PR and investigate